pkgs/by-name/fa/fastcap/package.nix

```diff
@@ -0,0 +1,88 @@
+{ stdenv
+, fetchzip
+, ghostscript
+, lib
+, texlive
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fastcap";
+  version = "2.0-18Sep92";
+
+  src = fetchzip {
+    url = "http://www.rle.mit.edu/cpg/codes/fastcap-${version}.tgz";
+    hash = "sha256-fnmC6WNd7xk8fphxkMZUq2+Qz+2mWIP2lvBUBAmUvHI";
+    stripRoot=false;
+  };
+
+  patches = [
+    ./patches/fastcap-mulGlobal-drop_conflicting_lib.patch
+    ./patches/fastcap-mulSetup-add_forward_declarations.patch
+  ];
+
+  nativeBuildInputs = [
+    ghostscript
+    texlive
+  ];
+
+  postPatch = ''
+    substituteInPlace ./doc/Makefile \
+      --replace '/bin/rm' 'rm'
+
+    for f in "doc/*.tex" ; do
+      sed -i -E 's/\\special\{psfile=([^,]*),.*scale=([#0-9.]*).*\}/\\includegraphics[scale=\2]{\1}/' $f
+      sed -i -E 's/\\psfig\{figure=([^,]*),.*width=([#0-9.]*in).*\}/\\includegraphics[width=\2]{\1}/' $f
+      sed -i -E 's/\\psfig\{figure=([^,]*),.*height=([#0-9.]*in).*\}/\\includegraphics[height=\2]{\1}/' $f
+      sed -i -E 's/\\psfig\{figure=([^,]*)\}/\\includegraphics{\1}/' $f
+    done
+
+    for f in "doc/mtt.tex" "doc/tcad.tex" "doc/ug.tex"; do
+      sed -i -E 's/^\\documentstyle\[(.*)\]\{(.*)\}/\\documentclass[\1]{\2}\n\\usepackage{graphicx}\n\\usepackage{robinspace}/' $f
+      sed -i -E 's/\\setlength\{\\footheight\}\{.*\}/%/' $f
+      sed -i -E 's/\\setstretch\{.*\}/%/' $f
+    done
+  '';
+
+  dontConfigure = true;
+
+  makeFlags = [
+    "CC=gcc"
+    "RM=rm"
+    "SHELL=sh"
+    "all"
+  ];
+
+  postBuild = ''
+    make manual
+    pushd doc
+    find -name '*.dvi' -exec dvipdf {} \;
+    popd
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/
+    mv bin $out/bin
+    rm $out/bin/README
+
+    mkdir -p $out/share/doc/fastcap
+    cp doc/*.pdf $out/share/doc/fastcap
+
+    mkdir -p $out/share/fastcap
+    mv examples $out/share/fastcap
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Multipole-accelerated capacitance extraction program";
+    longDescription = ''
+      Fastcap is a capacitance extraction program based on a
+      multipole-accelerated algorithm.'';
+    homepage = "https://www.rle.mit.edu/cpg/research_codes.htm";
+    license = licenses.mit; # see "src/fastcap.c"
+    maintainers = with maintainers; [ fbeffa ];
+    platforms = intersectLists (platforms.linux) (platforms.x86_64 ++ platforms.x86);
```

Comment:
This feels like it would've been reasonable to be able to build on darwin and bsd as well, but we would have to swap out the sed statements or something. Upstream only seem to specify `unix`. Is there a reason you're limiting it to `x86_64` and `x86`?